### PR TITLE
add logic to skip test for unspported images

### PR DESCRIPTION
**Pull request by feng-j678**
[x] add code logic to skip test for unspported images and unsuported agent version:
 examples of unsupported images:   
debian debian-11 11 0.20230124.1270
  redhat rhel-arm64 8_6-arm64 8.6.2022060901
  debian debian-11 11-backports 0.20221219.1234
  microsoftcblmariner cbl-mariner cbl-mariner-2-arm64 2.20230126.01
  suse sles-15-sp5 gen2 2024.02.07

skip debian/mariner:
image
image

### CODE REVIEW

File: microsoft/testsuites/vm_extensions/linux_patch_extension.py [hunk 1]
```diff
@@ -14,17 +14,57 @@
...
 from lisa.sut_orchestrator.azure.common import (
     get_compute_client,
     get_node_context,
     wait_operation,
 )
 from lisa.sut_orchestrator.azure.platform_ import AzurePlatform
+from lisa.util import SkippedException, UnsupportedDistroException
+
+
+def _is_os_unsupported(self, node: Node) -> None:
```
Comment (LiliDeng) [inline microsoft/testsuites/vm_extensions/linux_patch_extension.py]: no need self parameter

File: microsoft/testsuites/vm_extensions/linux_patch_extension.py [hunk 2]
```diff
@@ -14,17 +14,57 @@
...
+    }
+
+    arch = node.os.get_kernel_information().hardware_platform  # type: ignore
+    if arch == "aarch64":
+        unsupported_versions = unsupported_versions_arm64
+    else:
+        unsupported_versions = unsupported_versions_x86_64
+
+    # Extract Debian version from the complete version string
+    debian_version = node.os.information.version.major.split()[1]
```
Comment (LiliDeng) [inline microsoft/testsuites/vm_extensions/linux_patch_extension.py]: node.os.information.version.major doesn't support split, remove the debug line

File: microsoft/testsuites/vm_extensions/linux_patch_extension.py [hunk 3]
```diff
@@ -14,17 +14,57 @@
...
+    microsoftcblmariner cbl-mariner cbl-mariner-2-arm64 2.20230126.01
+    suse sles-15-sp5 gen2 2024.02.07
+    """
+    unsupported_versions_x86_64: Dict[type, List[str]] = {
+        Debian: [10, 11],
+        SLES: [15],
+        CBLMariner: [2],
+    }
+    unsupported_versions_arm64: Dict[type, List[str]] = {
+        Redhat: [8_6],
```
Comment (LiliDeng) [inline microsoft/testsuites/vm_extensions/linux_patch_extension.py]: Redhat: [8, 9],

File: microsoft/testsuites/vm_extensions/linux_patch_extension.py [hunk 4]
```diff
@@ -14,17 +14,57 @@
...
+    if arch == "aarch64":
+        unsupported_versions = unsupported_versions_arm64
+    else:
+        unsupported_versions = unsupported_versions_x86_64
+
+    # Extract Debian version from the complete version string
+    debian_version = node.os.information.version.major.split()[1]
+
+    for distro, version_list in unsupported_versions.items():
+        if isinstance(node.os, distro) and debian_version in version_list:
```
Comment (LiliDeng) [inline microsoft/testsuites/vm_extensions/linux_patch_extension.py]: `if isinstance(node.os, distro) and node.os.information.version.major in version_list:`

File: microsoft/testsuites/vm_extensions/linux_patch_extension.py [hunk 5]
```diff
@@ -14,17 +14,57 @@
...
+
+def _is_os_unsupported(self, node: Node) -> None:
+    """
+    Below specific Linux distro versions that are
+    not supported by linux patch extension.
+    debian debian-11 11 0.20230124.1270
+    redhat rhel-arm64 8_6-arm64 8.6.2022060901
+    debian debian-11 11-backports 0.20221219.1234
+    microsoftcblmariner cbl-mariner cbl-mariner-2-arm64 2.20230126.01
+    suse sles-15-sp5 gen2 2024.02.07
```
Comment (LiliDeng) [inline microsoft/testsuites/vm_extensions/linux_patch_extension.py]: no need include the details, list unsupported ones is good enough

**Comment by LiliDeng** [comment]
@feng-j678 could you please update this PR when you have time, thanks.

File: microsoft/testsuites/vm_extensions/linux_patch_extension.py (hunk 1 above)
Comment (feng-j678) [reply microsoft/testsuites/vm_extensions/linux_patch_extension.py]: logic changed

File: microsoft/testsuites/vm_extensions/linux_patch_extension.py (hunk 3 above)
Comment (feng-j678) [reply microsoft/testsuites/vm_extensions/linux_patch_extension.py]: logic changed

File: microsoft/testsuites/vm_extensions/linux_patch_extension.py (hunk 5 above)
Comment (feng-j678) [reply microsoft/testsuites/vm_extensions/linux_patch_extension.py]: logic changed

File: microsoft/testsuites/vm_extensions/linux_patch_extension.py (hunk 2 above)
Comment (feng-j678) [reply microsoft/testsuites/vm_extensions/linux_patch_extension.py]: logic changed

File: microsoft/testsuites/vm_extensions/linux_patch_extension.py (hunk 4 above)
Comment (feng-j678) [reply microsoft/testsuites/vm_extensions/linux_patch_extension.py]: logic changed

**Review by LiliDeng — APPROVED** [review]

**Pull request merged**